components/AuthForm/index.js

```diff
@@ -0,0 +1,149 @@
+import { useState, useRef } from "react";
+import { signIn } from "next-auth/react";
+import { useRouter } from "next/router";
+
+import { StyledCard } from "../StyledComponents/StyledCard";
+import { StyledForm, StyledFormInput, StyledFormLabel } from "../AddEntryForm";
+import {
+  StyledSettingsButton,
+  StyledSubmitButton,
+} from "../StyledComponents/StyledButtons";
+import styled from "styled-components";
+
+function AuthForm() {
+  const [registered, setRegistered] = useState(false);
+  const usernameInputRef = useRef();
+  const passwordInputRef = useRef();
+  const [loginError, setLoginError] = useState({
+    hasError: false,
+    errorMessage: "",
+  });
+
+  const [isLogin, setIsLogin] = useState(true);
```

Comment:
If I'm understanding what's happening correctly, then I'd probably prefer the naming here to reflect what `isLogin` does - something like `loginProcess` or similar. As it stands, `isLogin` is a little bit difficult to predict, which makes the code more difficult to understand.